### DISCUSSION TITLE
I just added styled components to my library legend-motion

### DISCUSSION THREAD

**Discussion by jmeistrich** [nativewind/nativewind#51] [Show and tell]
First of all I'm really happy this is even possible, so thanks a ton for this library 😀.

I added a set of components to my library using `styled` to make animatable styled components. I'd like to make sure with you that I'm doing it right. It just wraps the motion component around the `styled` component and it seems to work great: https://github.com/LegendApp/legend-motion/blob/main/src/styled.ts. It uses `tailwindcss-react-native` as a peerDependency so it's only used if imported from `@legendapp/motion/styled`.

My understanding from your fix to https://github.com/marklawlor/tailwindcss-react-native/issues/42 is that `styled` components should work fine whether the babel plugin is `compileOnly` or not?

I have docs up here (https://legendapp.com/dev/motion/tailwind-css/) just deferring to your installation steps.

Does that all look good? Is there anything you'd suggest I change? Are there specific installation steps or caveats that I should add?

**Comment by marklawlor** [comment]
That looks good! 

I haven't fully explored the option for 3rd party libraries supporting styled components and there are some API changes I can probably do to make things a bit more frictionless. 

For example, in the future you could provide a more seamless experience and remove the steps for  `configureMotion({ styled });` and the styled exports by detecting if the component is within the  `TailwindContext`.

You could technically do this now, as the `TailwindContext` is exported, but it's not documented as its API is still a bit unstable. It also includes the hooks for media queries so you might get some unnecessary renders. I'm refactoring some stuff atm, but I'll look into splitting out "platform" into its own context for easy detection and better integration.

**Reply by jmeistrich** [reply]
The reason for the styled exports is I don't want the whole library to depend on tailwindcss-react-native even if the user isn't using it, so the user can choose the styled version and take the dependency if they choose the styled export. Maybe there's a better way to do that?

The `configureMotion({ styled })` doesn't actually do anything in the library, it's just to force the user to import `styled`. If it's not imported, it's tree shaken out (because I have it as a peerDependency) and crashes when creating the components. This is my first open source library so I'm still learning about package structure, but I'm guessing there's a better way to solve that...

**Reply by marklawlor** [reply]
That all makes sense, in that case I'd leave it how it is. I think your using the best solution :+1:

**Reply by jmeistrich** [reply]
Ok, great! Thanks for the feedback 😀

**Comment by danstepanov** [comment]
Closing this discussion, if this is still relevant, feel free to open a new one.
